README.md

```diff
@@ -0,0 +1,695 @@
+
+```coffeescript
+[1..3]
+  .map((x) -> x * x)
+  .concat([10..12])
+  .filter((x) -> x < 11)
+  .reduce((x, y) -> x + y)
+```
+
+When calling functions, always use parentheses.
+```
+'no_implicit_parens'
+```
+
+```coffeescript
+baz()
+baz(12)
+
+brush.ellipse(x: 10, y: 20) # Braces can be omitted or included for readability
+
+foo(4).bar(8)
+
+obj.value(10, 20) / obj.value(20, 10)
+
+print(inspect(value))
+
+new Tag(new Value(a, b), new Arg(c))
+```
+
+<a name="strings"></a>
+## Strings
+
+Use string interpolation instead of string concatenation:
+```
+'no_interpolation_in_single_quotes': {
+  'level': 'error'
+}
+```
+
+```coffeescript
+"this is an #{adjective} string" # Yes
+"this is an " + adjective + " string" # No
+```
+
+Prefer single quoted strings (`''`) instead of double quoted (`""`) strings, unless features like string interpolation are being used for the given string.
+```
+'no_unnecessary_double_quotes': {
+  'level': 'error'
+}
+```
+
+<a name="conditionals"></a>
+## Conditionals
+
+Favor `if` over `unless` for negative conditions.
+
+```coffeescript
+# No
+result = y unless x
+
+# Yes
+result = y if not x
+```
+
+
+```coffeescript
+  # Yes
+  if true
+    ...
+  else
+    ...
+
+  # No
+  unless false
+    ...
+  else
+    ...
+```
+
+Multi-line if/else clauses should use indentation:
+
+```coffeescript
+  # Yes
+  if true
+    ...
+  else
+    ...
+
+  # No
+  if true then ...
+  else ...
+```
+
+<a name="looping_and_comprehensions"></a>
+## Looping and Comprehensions
+
+For arrays, try to use native Array.prototype methods
+
+```coffeescript
+  # No
+  result = (item.name for item in array)
+
+  # No
+  results = []
+  for item in array
+    results.push item.name
+
+  # Yes
+  result = array.map((item) -> item.name)
+```
+
+To filter:
+
+```coffeescript
+result = items.filter((item) -> item.name is 'test')
+```
+
+To iterate over the keys and values of objects:
+
+```coffeescript
+object =
+  one: 1,
+  two: 2
+alert("#{key} = #{value}") for key, value of object
+```
+
+<a name="extending_native_objects"></a>
+## Extending Native Objects
+
+Do not modify native objects.
+
+For example, do not modify `Array.prototype` to introduce `Array#forEach`.
+
+<a name="exceptions"></a>
+## Exceptions
+
+Do not suppress exceptions.
+
+<a name="annotations"></a>
+## Annotations
+
+Use annotations when necessary to describe a specific action that must be taken against the indicated block of code.
+
+Write the annotation on the line immediately above the code that the annotation is describing.
+
+The annotation keyword should be followed by a colon and a space, and a descriptive note.
+
+```coffeescript
+  # FIXME: The client's current state should *not* affect payload processing.
+  resetClientState()
+  processPayload()
+```
+
+If multiple lines are required by the description, indent subsequent lines with two spaces:
```

Comment:
Ah yeah that sounds fine. We actually decided to use `###` notation for block comments. Although I don't know what counts as a block. > 1 line?